core/src/main/java/org/eclipse/hono/util/DataVolume.java

```diff
@@ -0,0 +1,142 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.util;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Optional;
+
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSetter;
+
+/**
+ * Data volume definition of the tenant resource limits.
+ */
+@JsonInclude(Include.NON_DEFAULT)
+public class DataVolume {
+
+    private Instant effectiveSince;
+
+    @JsonProperty(TenantConstants.FIELD_MAX_BYTES)
+    private long maxBytes = TenantConstants.DEFAULT_MAX_BYTES;
+
+    @JsonProperty(TenantConstants.FIELD_PERIOD)
+    private DataVolumePeriod period;
+
+    /**
+     * Gets the point in time on which the data volume limit came into effect.
+     *
+     * @return The instant as an ISO string or 
+     *         {@code null} if not set.
+     */
+    @JsonGetter(TenantConstants.FIELD_EFFECTIVE_SINCE)
+    public String getEffectiveSinceAsString() {
```

Comment:
we should be able to use a standard Jackson serializer as introduced in #1374

Reply:
Also, take a look at the recent changes done in #1463. I think it should be aligned with that.